File: P202 Happy Number/Solution.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        unique_number_set = set()
        while n not in unique_number_set:
            unique_number_set.add(n)
            value = 0
            while n > 0:
                value += (n%10)*(n%10)
                n /= 10
            n = value
        return n == 1

File: P202 Happy Number/Solution.kt
class Solution {
    fun isHappy(n: Int): Boolean {
        var num = n
        val uniqueNumberSet = hashSetOf<Int>()
        while(uniqueNumberSet.add(num)) {
            var value = 0
            while(num > 0) {
                value += Math.pow(1.0 * num % 10, 2.0).toInt()
                num /= 10
            }
            num = value
        }
        return num == 1
    }
}